### DISCUSSION TITLE
KafkaConnect trying to mount volume "docker-credentials" with key .dockerconfigjson on non-OpenShift build...is that correct?

### DISCUSSION THREAD

**Discussion by mocan55** [strimzi/strimzi-kafka-operator#9359] [Q&A] [answered]
I looked at [cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectBuild.java].  It appears that this file is only required if using OpenShift.  I am not using OpenShift and specified spec.build.output.type as "docker".  Should my build try to mount a volume with this file included.  It is causing the build to fail.  This is my sanitized KafkaConnect yaml. 

apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaConnect
metadata:
   name: {{ .Values.environment }}
   namespace: sg-{{ .Values.environment }}
   annotations:
     strimzi.io/use-connector-resources: "true"
spec:
   version: 3.4.0
   replicas: 1
   bootstrapServers: {{ .Values.environment }}-kafka-bootstrap:9093
   tls:
     trustedCertificates:
       - secretName: {{ .Values.environment }}-cluster-ca-cert
          certificate: ca.crt
   config:
     group.id: {{ .Values.environment }}
     offset.storage.topic: {{ .Values.environment }}-offsets
     config.storage.topic: {{ .Values.environment }}-configs
     status.storage.topic: {{ .Values.environment }}-status
     config.storage.replication.factor: -1
     offset.storage.replication.factor: -1
     status.storage.replication.factor: -1
  build:
     output:
       type: docker
       image: harbor.boxops.agsadmin.com/apps/sg-kafka-connect:latest
       pushSecret: robot-harbor
     plugins:
       - name: debezium-postgres-connector
         artifacts:
           - type: jar
             url: https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/2.4.0.Final/debezium-connector-postgres- 
 2.4.0.Final.jar
       - name: snowflake-kafka-connector
         artifacts:
           - type: jar
             url: https://repo1.maven.org/maven2/com/snowflake/snowflake-kafka-connector/2.1.0/snowflake-kafka-connector-2.1.0.jar

**Comment by scholzj** [comment]
No, it is required anytime you push to some container registry that requires authentication. Both on OpenShift but also outside of OpenShift.

**Reply by mocan55** [reply]
OK. Are these for accessing our private repository or something else?  The reason I ask is that we already supply that as "pushSecret:"

**Reply by scholzj** [reply]
The secret in the `.spec.build` section is used to push the newly built image to the registry specified in `.spec.build.image`. The one in `.spec.template.pod` is used for pulling the image in the Connect pods.

Just to be clear -> both are optional, you need them only if authentication is needed to push the new container image in the first case or pull it in the second case.

**Comment by mocan55** [comment]
Ok.   I specified the key in both but still get the mount volume error.

MountVolume.SetUp failed for volume "docker-credentials" : references
non-existent secret key: .dockerconfigjson

On Fri, Nov 17, 2023 at 5:24 PM Jakub Scholz ***@***.***>
wrote:

> The secret in the .spec.build section is used to push the newly built
> image to the registry specified in .spec.build.image. The one in
> .spec.template.pod is used for pulling the image in the Connect pods.
>
> Just to be clear -> both are optional, you need them only if
> authentication is needed to push the new container image in the first case
> or pull it in the second case.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/strimzi/discussions/9359#discussioncomment-7603635>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2K43HVZSKILNF3FJM6IS5LYE7PZLAVCNFSM6AAAAAA7QAWFH6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMBTGYZTK>
> .
> You are receiving this because you authored the thread.Message ID:
> <strimzi/strimzi-kafka-operator/repo-discussions/9359/comments/7603635@
> github.com>
>


-- 
[image: Profile picture] Pat McNeace
DevSecOps Engineer
855.895.8951 [office]  //  706.622.4263 [fax]
LinkedIn <https://www.linkedin.com/company/securegive>  //  Facebook
<https://www.facebook.com/securegive>  //  Website
<https://www.securegive.com?utm_source=signaturesatori&utm_medium=email&utm_campaign=signature>
[image: SecureGive]
------------------------------

**Reply by scholzj** [reply, marked as answer]
And does your secret contain that key? The `type: kubernetes.io/dockerconfigjson` secrets  with Docker registry credentials should normally have this key.

**Comment by mocan55** [comment]
I transfer the repository creds to the cluster as username and key.  I will
create a JSON file containing them as well.

On Mon, Nov 20, 2023 at 9:41 AM Jakub Scholz ***@***.***>
wrote:

> And does your secret contain that key? The type:
> kubernetes.io/dockerconfigjson secrets with Docker registry credentials
> should normally have this key.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/strimzi/discussions/9359#discussioncomment-7620299>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2K43HQWRLC2WJ7255D6MKDYFNTYFAVCNFSM6AAAAAA7QAWFH6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRQGI4TS>
> .
> You are receiving this because you authored the thread.Message ID:
> <strimzi/strimzi-kafka-operator/repo-discussions/9359/comments/7620299@
> github.com>
>


-- 
[image: Profile picture] Pat McNeace
DevSecOps Engineer
855.895.8951 [office]  //  706.622.4263 [fax]
LinkedIn <https://www.linkedin.com/company/securegive>  //  Facebook
<https://www.facebook.com/securegive>  //  Website
<https://www.securegive.com?utm_source=signaturesatori&utm_medium=email&utm_campaign=signature>
[image: SecureGive]
------------------------------

**Comment by mocan55** [comment]
Thanks, Jakub!  We usually use type.opaque for secrets. Changed to
type.registry.  The problem was fixed.

On Mon, Nov 20, 2023 at 10:00 AM Pat McNeace ***@***.***>
wrote:

> I transfer the repository creds to the cluster as username and key.  I
> will create a JSON file containing them as well.
>
> On Mon, Nov 20, 2023 at 9:41 AM Jakub Scholz ***@***.***>
> wrote:
>
>> And does your secret contain that key? The type:
>> kubernetes.io/dockerconfigjson secrets with Docker registry credentials
>> should normally have this key.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/orgs/strimzi/discussions/9359#discussioncomment-7620299>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A2K43HQWRLC2WJ7255D6MKDYFNTYFAVCNFSM6AAAAAA7QAWFH6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRQGI4TS>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> <strimzi/strimzi-kafka-operator/repo-discussions/9359/comments/7620299@
>> github.com>
>>
>
>
> --
> [image: Profile picture] Pat McNeace
> DevSecOps Engineer
> 855.895.8951 [office]  //  706.622.4263 [fax]
> LinkedIn <https://www.linkedin.com/company/securegive>  //  Facebook
> <https://www.facebook.com/securegive>  //  Website
> <https://www.securegive.com?utm_source=signaturesatori&utm_medium=email&utm_campaign=signature>
> [image: SecureGive]
> ------------------------------
>


-- 
[image: Profile picture] Pat McNeace
DevSecOps Engineer
855.895.8951 [office]  //  706.622.4263 [fax]
LinkedIn <https://www.linkedin.com/company/securegive>  //  Facebook
<https://www.facebook.com/securegive>  //  Website
<https://www.securegive.com?utm_source=signaturesatori&utm_medium=email&utm_campaign=signature>
[image: SecureGive]
------------------------------
